src/qibo/tomography/gate_set_tomography.py

```diff
@@ -278,7 +292,22 @@ def GST(
 
     for gate in gate_set:
         if gate is not None:
-            init_args = signature(gate).parameters
+            if isinstance(gate, tuple):
+                init_args = signature(gate[0]).parameters
+                params = gate[1]
+                gate_class = gate[0]
+            else:
+                init_args = signature(gate).parameters
+                params = []
+                gate_class = gate
+
+            angle_names = [
+                name for name in init_args if name in {"theta", "phi", "lam"}
+            ]
+            angle_values = {}
+            for name, value in zip(angle_names, params):
+                angle_values[name] = value
```

Comment:
```suggestion
            if isinstance(gate, tuple):
                angles = ["theta", "phi", "lam"]
                gate, params = gate
                angle_values = dict(zip(angles[:len(params)], params))
            else:
                angle_values = {}
            init_args = signature(gate).parameters
```

this should do the same I believe